pagopa/io-app
# feat: [IOCOM-856] Add CTAs to `MessageDetailsScreen`

## Short description
This PR adds the CTAs extracted from message markdown (defined as front matter)

<details open><summary>Details</summary>
<p>

| iOS  | iOS |
| - | - |
| <img src="https://github.com/pagopa/io-app/assets/29163287/a47be6b4-c3e1-4799-aaec-1f819058c672" width="300"/> | <img src="https://github.com/pagopa/io-app/assets/29163287/e59a76c3-3590-4af7-80f0-b1c2cb42e296" width="300"/> |

| Android | Android |
| - | - |
| <img src="https://github.com/pagopa/io-app/assets/29163287/e970a50c-32f9-4b5c-8c35-7344b2c463a7" width="300"/> | <img src="https://github.com/pagopa/io-app/assets/29163287/260d7bb0-47c7-4d22-9e37-f148cc21a1eb" width="300"/> |

</p>
</details> 

## List of changes proposed in this pull request
- added FooterCTAs component to extract the CTAs
- added ExtractedCTAs component to render the extracted CTAs
- added usePNOptInMessage hook to check if it's an optin message

## How to test
Using `io-dev-api-server`, generate a message with CTAs. Activate experimental ds and open a message. Check that CTAs are displayed correctly.
